components/layout/client.coffee

```diff
@@ -54,7 +54,7 @@ initGlobalBlockRouting = ->
     /^\/reset(\/\w+)?$/
     /^\/confirm(\/\w+)?$/
     /^\/register(\/\w+)?$/
-    /^\/welcome(\/\w+)?$/
+    /^\/welcome(\/.*)?$/
```

Comment:
`^\/welcome(\/(\w|-)+)?`

Would be more accurate for URLs.

You can also change all of these to that.